src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -11745,6 +11745,105 @@ def str_len(self, **kwargs: Any) -> "SnowflakeQueryCompiler":
             self._modin_frame.apply_snowpark_function_to_data_columns(length)
         )
 
+    def str_slice(
+        self,
+        start: Optional[int] = None,
+        stop: Optional[int] = None,
+        step: Optional[int] = None,
+    ) -> "SnowflakeQueryCompiler":
+        """
+        Slice substrings from each element in the Series or Index.
+
+        Parameters
+        ----------
+        start : int, optional
+            Start position for slice operation.
+        stop : int, optional
+            Stop position for slice operation.
+        step : int, optional
+            Step size for slice operation.
+
+        Returns
+        -------
+        SnowflakeQueryCompiler representing result of the string operation.
+        """
+        if step == 0:
+            raise ValueError("slice step cannot be zero")
+
+        def output_col(
+            col_name: ColumnOrName,
+            start: Optional[int],
+            stop: Optional[int],
+            step: Optional[int],
+        ) -> SnowparkColumn:
+            if step is None:
+                step = 1
+            col_len_exp = builtin("len")(col(col_name))
+
+            if start is None:
+                if step < 0:
+                    start_exp = col_len_exp
+                else:
+                    start_exp = pandas_lit(1)
+            elif start < 0:
+                if step < 0:
+                    start_exp = builtin("greatest")(
```

Comment:
```suggestion
                    start_exp = greatest(
```
Same as `length`, you'll need to import this from `snowflake.snowpark.functions`.